### PR TITLE
[ME-2597] Replace TUN iface library in vpnlib

### PR DESCRIPTION
**Pull request by adrianosela**
## [[ME-2597](https://mysocket.atlassian.net/browse/ME-2597)] Replace TUN iface library in vpnlib

Replace network iface library in `vpnlib` package for a fork of the original library that adapts it to work for TUN ifaces on windows.

Fixes # (issue)
- https://mysocket.atlassian.net/browse/ME-2597

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

This was tested locally on a windows machine.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code

[ME-2597]: https://mysocket.atlassian.net/browse/ME-2597?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by atoonk — APPROVED** [review]
This is ready to go as far as I'm concerned. Now proper support for windows.

File: cmd/client/vpn/vpn.go [hunk 1]
```diff
@@ -77,9 +79,37 @@ var clientVpnCmd = &cobra.Command{
...
+			err := wintundll.Ensure(
+				wintundll.WithDownloadURL("https://www.wintun.net/builds/wintun-0.14.1.zip"),
+				wintundll.WithDownloadTimeout(time.Second*10),
+				wintundll.WithDllPathToEnsure(`C:\Windows\System32\wintun.dll`),
+			)
+			if err != nil {
+				return fmt.Errorf("Error ensuring Wintun driver is installed: %w", err)
+			}
+
+			ifName := "border0VPN"
```
Comment (adrianosela) [inline cmd/client/vpn/vpn.go:97]: This can go directly in line below

File: cmd/client/vpn/vpn.go [hunk 2]
```diff
@@ -195,7 +227,21 @@ var clientVpnCmd = &cobra.Command{
...
-		if err = vpnlib.AddRoutesToIface(iface.Name(), ctrl.Routes); err != nil {
+		// check if we're running on Windows , if so we need to sleep for a few seconds
+		// Seems like it's slow to update the routing table on Windows
+		// if we don't wait, windows will use the wrong "Interface" for the route.
+		// It seems to then choose the old default gateway, which is not what we want
+		// So we wait a few seconds, and then add the routes, it then picks the VPN interfaces as the correct interface
+		if runtime.GOOS == "windows" {
+			fmt.Printf("Adding VPN routes")
+			// for loop with one 500ms sleep each
+			for i := 0; i < 10; i++ {
```
Comment (adrianosela) [inline cmd/client/vpn/vpn.go:238]: This is adding a 5 second delay for unclear and undocumented reasons.

File: internal/vpnlib/dnsinfo_windows.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	"fmt"
+	"runtime"
+
+	"golang.zx2c4.com/wireguard/windows/tunnel/winipcfg"
+)
+
+// GetDnsServers returns a list of all active resolvers used by the system
+func GetDnsServers() ([]string, error) {
+	switch runtime.GOOS {
+	case "windows":
```
Comment (adrianosela) [inline internal/vpnlib/dnsinfo_windows.go:16]: Redundant case.  Compiled for windows will always have GOOS windows -- but this is fine.

**Pull request merged**